### ISSUE TITLE
Option to not fail based on severity

### ISSUE THREAD

**Issue by PedroArvela** [hadolint/hadolint#607]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

We have a CI pipeline that succeeds or fails based on hadolint's return code. Errors of type `info` and `style` should not cause hadolint to fail. We would like to be able to set what severities would cause the linter to fail.

### Actual behavior

Linter failure can only be globally disabled, not disabled by severity type. Ignores can only be added per individual codes, not by severity type.

### Steps to reproduce the behavior

Create a Dockerfile with an `info` level severity.

Run `hadolint Dockerfile`

The return code will be `1` instead of `0` even though the severity type of the error is `info`.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v2.2.0-0-ga34e5f3
```

**Dockerfile** (if relevant)

```Dockerfile
FROM debian

# Example dockerfile with an info level severity lint error
RUN echo hello
RUN echo world
```

**Additional environment details (OS, stack version, etc.)**

**Comment by lorenzo** [comment]
thanks, this is a great idea. If you want to help implement it, I can offer help and guidance

**Comment by holgergp** [comment]
I'd like to do this, but I won't find the time to do this before this before the weekend. If someone has time before that, please go for it.
I have only basic Haskell knowledge though, I've never done anything beyond katas, so this might get a little rocky.
But it would be awesome to improve my skills here. 😀

**Comment by lorenzo** [comment]
@holgergp happy to offer guidance, let me know if you have questions. I can come up with a high-level plan for you later today
